File: example/rabbitmoproducer/src/main/java/com/example/rabbitmoproducer/MsgReceiver.java
package com.example.rabbitmoproducer;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.amqp.rabbit.annotation.RabbitHandler;
import org.springframework.amqp.rabbit.annotation.RabbitListener;
import org.springframework.stereotype.Component;

/**
 * @author wangxingxiang
 * @Description
 * @date 2019/9/19 18:06
 */
/*@Component
@RabbitListener(queues = {RabbitConfig.QUEUE_A,RabbitConfig.QUEUE_B})
public class MsgReceiver {
    private final Logger logger = LoggerFactory.getLogger(this.getClass());

    @RabbitHandler
    public void process(String content) {
        logger.info("接收处理队列A当中的消息： " + content);
    }


}*/

File: example/demokt/src/main/kotlin/com/example/demo/MsgReceiver.kt
package com.example.demo

import org.slf4j.LoggerFactory
import org.springframework.amqp.rabbit.annotation.RabbitHandler
import org.springframework.amqp.rabbit.annotation.RabbitListener
import org.springframework.stereotype.Component

/**
 *
 *
 * @author wangxingxiang
 * @Description
 * @date 2019/9/19 18:47C
 */
@Component
@RabbitListener(queues = ["QUEUE_A", "QUEUE_B","QUEUE_C"])
class MsgReceiver {
    private val logger = LoggerFactory.getLogger(this.javaClass)

    @RabbitHandler
    fun process(content: String) {
        logger.info("接收处理队列A当中的消息： $content")
    }
}